test/Streamly/Test/Prelude/Async.hs

```diff
@@ -56,6 +56,14 @@ main = hspec
         asyncOps $ prop "asyncly consM" . constructWithConsM S.consM sort
         asyncOps $ prop "asyncly (.:)" . constructWithCons (S..:)
         asyncOps $ prop "asyncly (|:)" . constructWithConsM (S.|:) sort
+        prop "asyncSingleThreaded" $
+            constructfromAsyncSingleThread
+            (S.fromList [1,2,3,4,5])
+            (S.fromList [6,7,8,9,10])
+            [1,2,3,4,5,6,7,8,9,10]
+
+        prop "asyncMultiThreaded" $
+            constructfromAsyncNThread 4 2 [2,4]
```

Comment:
This is already being at line 127 below using `parallelCheck`. We can remove it.